### PR TITLE
INGEST: Tests for Drop Processor

### PR DESCRIPTION
**Pull request by original-brownbear**
* UT for behavior of dropped callback
and drop processor
   * Moved drop processor to `server` project to enable this test (otherwise I would've had to add code for a dummy processor that returns `null` in the server tests for testing the callback ... moving the proc seemed more reasonable :))
* Simple IT
* Relates #32278

### CODE REVIEW

**Comment by original-brownbear** [comment]
@rjernst can you take a look here? :) I think we should at least have some basic tests  for this.

File: server/src/test/java/org/elasticsearch/ingest/IngestServiceTests.java [hunk 1]
```diff
@@ -812,6 +812,27 @@ public void testUpdatingStatsWhenRemovingPipelineWorks() {
         assertThat(ingestService.stats().getStatsPerPipeline(), not(hasKey("_id2")));
     }
 
+    public void testExecuteWithDrop() {
+        IngestService ingestService = createWithProcessors(Collections.singletonMap("drop", new DropProcessor.Factory()));
+        PutPipelineRequest putRequest = new PutPipelineRequest("_id",
+            new BytesArray("{\"processors\": [{\"drop\" : {}}]}"), XContentType.JSON);
```
Comment (jakelandis) [inline server/src/test/java/org/elasticsearch/ingest/IngestServiceTests.java]: minor: you may also want a second (mock) processor here, and assert that the second processor never executes.

**Comment by jakelandis** [comment]
LGTM, one minor optional suggestion.

**Comment by original-brownbear** [comment]
@jakelandis thanks! Suggestion applied :)

**Pull request merged**